ehsm_kms_service/quote_manager_apis.js

```diff
@@ -0,0 +1,270 @@
+const {
+    cryptographic_apis
+} = require('./apis')
+
+
+/**
+ * Verify that the parameter is string and whether it is required and the maximum length
+ * @param {object} param : Variables requiring validation
+ * @param {boolean} required : Whether the parameter is required
+ * @param {int} maxLength [defalut=undefined] : Maximum length of string, If maxLength is undefined, the length is not verified.
+ * @returns {boolean}
+ */
+function checkStringParam(param, required, maxLength) {
+    if (param == '' || param == undefined) {
+        if (required) {
+            return false
+        } else {
+            return true
+        }
+    } else {
+        if (typeof (param) == 'string') {
+            if (maxLength != undefined && param.length > maxLength) {
+                return false
+            } else {
+                return true
+            }
+        } else {
+            return false
+        }
+    }
+}
+
+/**
+ * check uuid format
+ * @param {String} uuid : a uuid string, eg. 0197ad2d-c4be-4948-996d-513c6f1e****
+ * @returns {boolean}
+ */
+function check_UUID_format(uuid) {
+    if (uuid != '' && uuid != undefined) {
+        if (!((/^([a-z\d]{8}-[a-z\d]{4}-[a-z\d]{4}-[a-z\d]{4}-[a-z\d]{12})$/).test(uuid))) {
+            return false
+        }
+    }
+    return true
+}
+
+/**
+ * Generate a quote of the eHSM-KMS core enclave
+ * @param {Object} res : response
+ * @param {Object} payload
+ *          ==> {r}challenge(String [1~1024]) : a string. eg. 'Y2hhbGxlbmdl'
+ * @returns {Object}
+ *          ==> {r}challenge(String) : a string. eg. 'Y2hhbGxlbmdl'
+ *          ==> {r}quote(String) : A quote for the eHSM-KMS core enclave format in BASE64 string. eg. 'AwACAAAAAAAHAAwAk5pB∗∗∗'
+ */
+const generateQuote = async (res, payload, action) => {
+    try {
+        const { challenge } = payload
+        if (checkStringParam(challenge, true, CHALLENGE_LENGTH_MAX)) {
+            napi_res = napi_result(action, res, [challenge])
+            napi_res && res.send(napi_res)
+        } else {
+            res.send(_result(400, `challenge cannot be empty, must be string and length not more than ${CHALLENGE_LENGTH_MAX}`))
+            return
+        }
+    } catch (e) {
+        console.info('generateQuote :: ', e)
+        res.send(_result(500, 'Server internal error, please contact the administrator.'))
+        return
+    }
+}
+
+/**
+ * Users are expected already got a valid DCAP format QUOTE. And it could use this API to send it to eHSM-KMS to do a quote verification.
+ * @param {Object} res : response
+ * @param {String} appid : appid of user
+ * @param {Object} DB : database controller
+ * @param {Object} payload
+ *          ==> {r}quote(String [1~10924]) : A valid DCAP quote in BASE64 string. eg. 'AwACAAAAAAAHAAwAk5pB∗∗∗'
+ *          ==> {r}nonce(String [1~1024]) : A nonce in BASE64 string. eg. 'bm9uY2U='
+ *          ==> {r}policyId(String [36]) : quote policy ID. eg. '326f2049-56ab-474f-a1c0-516de230****'
```

Comment:
why policyid is required?

Reply:
Can we still support the case to verify Quote only but not check mrenclave and mrsigner?

Reply:
need use policyid to find mr_enclave&mr_signer to verification quote.

Reply:
the current logic, these two values must be verified

Reply:
we can do it later, such as add a new flag to tell ehsm whether to do the policy verification. 

it's optional, not required in this PR.

Reply:
if policyid is undefined, we can skip the comparisons on mr_enclave&mr_signer.